src/cmd/run.go

```diff
@@ -283,10 +277,14 @@ func runCommand(container string,
 
 	execArgs = append(execArgs, []string{
 		container,
-		"capsh", "--caps=", "--", "-c", "exec \"$@\"", "/bin/sh",
+		"capsh", "--caps=", "--", "-c", commandLauncher, "toolbox",
 	}...)
 
-	execArgs = append(execArgs, command...)
+	if command != nil {
+		execArgs = append(execArgs, command...)
+	} else {
+		command = []string{"\"$SHELL\""}
```

Comment:
command variable cannot be null as it's accessed in the error messages after Podman returns.

Reply:
I'm not certain these error handling should be kept as is though, since it can be misleading. For example, if you enter a toolbox, try to launch a command which doesn't exist, and immediately exit the toolbox (with ctrl+d), you'll get "Error: command /bin/bash not found in container fedora-toolbox-32" (assuming your shell is /bin/bash and your toolbox container is named fedora-toolbox-32).